libr/include/r_vector.h

```diff
@@ -23,25 +23,25 @@ typedef struct r_vector_t {
 	size_t len;
 	size_t capacity;
 	size_t elem_size;
+	RVectorFree free;
+	void *free_user;
 } RVector;
 
-typedef struct r_pvector_t {
-	RVector v;
-	RPVectorFree free;
-} RPVector;
+// RPVector directly wraps RVector for type safety
+typedef struct r_pvector_t { RVector v; } RPVector;
 
 
 // RVector
 
-R_API void r_vector_init(RVector *vec, size_t elem_size);
+R_API void r_vector_init(RVector *vec, size_t elem_size, RVectorFree free, void *free_user);
```

Comment:
agreed

Reply:
Although Idk, since RVectorFree is defined to expect a user parameter, you would expect to be able to set it somewhere too.

Reply:
I think there are ways to remove that from the API, but if that takes too much time it's also ok as it is.

Reply:
Yeah there definitely are. I thought for example about simply rewriting r_pvector_free/clear to apply the free function on the pointers instead of pointers to the pointers, but that would also mean rewriting the freeing of the array and so on, which is more an internal thing of vector, so it becomes a bit hacky.
It's clearly a matter of opinion, but I would rather keep the user pointer if this means I can encapsulate the code better.
After all, RVector is the only data structure we have that does not necessarily carry only pointers, which makes it different from all the others anyway, so having the user pointer there is not a big deal imho in terms of consistency.